webapp/handlers.py

```diff
@@ -114,6 +120,16 @@ def clear_trailing():
 
     @app.before_request
     def prometheus_metrics():
+        # Accept-encoding counter
+        user_agent = user_agents.parse(flask.request.headers.get("User-Agent"))
+
+        accept_encoding_counter.labels(
+            accept_encoding=flask.request.headers.get("Accept-Encoding"),
+            browser_family=user_agents.browser.family,
```

Comment:
It seems that this make the k8s check fail.
You might want to check if the user agent is defined

https://sentry.is.canonical.com/canonical/snapcraft-io/issues/16317/?query=is%3Aunresolved